consensus/consensus_fbft.go

```diff
@@ -0,0 +1,30 @@
+package consensus
+
+import "sync"
+
+type LockedFBFTPhase struct {
+	mu     sync.Mutex
+	phrase FBFTPhase
+}
+
+func NewLockedFBFTPhase(initialPhrase FBFTPhase) *LockedFBFTPhase {
+	return &LockedFBFTPhase{
+		phrase: initialPhrase,
+	}
+}
+
+func (a *LockedFBFTPhase) Set(phrase FBFTPhase) {
+	a.mu.Lock()
+	a.phrase = phrase
+	a.mu.Unlock()
+}
```

Comment:
you might want to consider using "defer"

Reply:
Be aware, that `defer` is not free, it has overhead. It is useful when we want to reduce code complexity or add panic catch possibility.  We are not interested in catching panics, and there is no complexity to reduce. I don't think that adding `defer` just in case is a good idea. But if code gets somehow more complex, i will add defer for sure.